src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpSimplifyTypeNamesDiagnosticAnalyzer.cs

```diff
@@ -30,7 +31,38 @@ internal sealed class CSharpSimplifyTypeNamesDiagnosticAnalyzer
                 SyntaxKind.SimpleMemberAccessExpression,
                 SyntaxKind.QualifiedCref);
 
-        protected override void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
+        protected override bool IsIgnoredCodeBlock(ref CodeBlockAnalysisContext context)
+        {
+            return context.CodeBlock.IsKind(
```

Comment:
consider just passing the codeblock along instead of the context?

Reply:
Updated and documented